n3fit/src/n3fit/backends/keras_backend/MetaModel.py

```diff
@@ -353,14 +354,12 @@ def get_replica_weights(self, i_replica):
             dict
                 dictionary with the weights of the replica
         """
-        NN_weights = [
-            tf.Variable(w, name=w.name) for w in self.get_layer(f"{NN_PREFIX}_{i_replica}").weights
-        ]
-        prepro_weights = [
-            tf.Variable(w, name=w.name)
-            for w in self.get_layer(f"{PREPROCESSING_PREFIX}_{i_replica}").weights
-        ]
-        weights = {NN_PREFIX: NN_weights, PREPROCESSING_PREFIX: prepro_weights}
+        weights = {}
+        for layer_type in [NN_LAYER, PREPROESSING_LAYER]:
+            weights[layer_type] = [
+                tf.Variable(w, name=w.name)
+                for w in get_layer_replica_weights(self.get_layer(layer_type), i_replica)
+            ]
```

Comment:
```suggestion
        weights = defaultdict(list)
        for layer_type in [NN_LAYER, PREPROCESSING_LAYER]:
            layer_ref = self.get_layer(layer_type)
            for w in get_layer_replica_weights(layer_ref, i_replica):
                weights[layer_type].append(tf.Variable(w, name=w.name))
```

This doesn't get called many times, right? (For the same number of lines I'd rather have a more explicit approach, otherwise the list comprehension might be useful)

Also, why do you need `tf.Variable` here if you are already doing `tf.Variable` in `get_layer_replica_weights`?
If needed, wouldn't it be better to do it in that function?

Reply:
I think you're right that `tf.Variable` is not needed here, so it just becomes a oneliner.